### ISSUE TITLE
Support pseudo states with params

### ISSUE THREAD

**Issue by tomrav** [wix/stylable-intelligence#179]
Example:

```css
.myClass {
  -st-states: myState( string( minLength(3), regex("^user"), contains(name) );
}

.myClass:myState(username);
```

#### completions:  
- [x] available types
- [x] available validators for a given type (allowing repeating validators)
#### signature hinting:
- [x] available validators for a given type
- [ ] inside each validator hint their arguments (currently limited to a single argument)

#### completions:
- [x] state name
- [x] for enum type, suggest possible parameter values
#### signature hinting:
- [x] show type signature with defined validators
 
> Also, make sure we don't break go to definition

**Comment by tempit** [comment]
Also make sure findRefs, rename still work, they may try to rename the state+param

**Comment by tomrav** [comment]
implemented in #193
